database/migrations/2017_12_20_173343_add_foreign_columns_to_users_table.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+use Illuminate\Support\Facades\Schema;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Database\Migrations\Migration;
+
+class AddForeignColumnsToUsersTable extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::table('users', function (Blueprint $table) {
+            $table->integer('team_id')->nullable()->unsigned();
+            $table->integer('owner_team_id')->nullable()->unsigned();
```

Comment:
This is redundant. The Team model already has a `user_id` that represents the owner.

Reply:
@fntneves being this a relationship [one-to-one](https://laravel.com/docs/5.5/eloquent-relationships#one-to-one), a team as one owner and a user can only be owner of one team, shouldn't both the tables have a foreign key?

Reply:
No, only the Team model should have a `user_id` column. The solution for a real *one-one* is to add a unique() constraint to `user_id` on `teams` table. Otherwise, it will be managed by the application.

Additionally, as we are allowing soft deletes in Team model, we should not force this on database.

Reply:
Ok then, my bad, this was enlightening as always, thank you :+1: